Dockerfile

```diff
@@ -1,24 +1,32 @@
 # This is a multi-stage image build.
 #
-# We first create a "builder" image and then create our final image by copying
-# things from the builder image.  The point is to avoid bloating the final
+# We first create two "builder" images:
+# 1. native-builder, for packages that can be installed for a different target platform (native build, native execution)
+# 2. emulated-builder, for packages that must be installed targeting the build platform (emulated build, native execution)
```

Comment:
I think this description needs some rewording and clarification. And in doing so, we might want to rename the builder images.

There's two intersecting things here, right?

1. build vs. target platform
2. if build == target or not

Names and commentary seem to assume that build != target?

But when build == target, then emulation is never in use and `native-builder` and `emulated-builder` don't make much sense as names? I think the use of "emulated" here is assuming the scenario of a Linux x86-64 build platform targeting aarch64.

Reply:
Yes thanks for breaking that down, I'll reword. As for names, `strictly-native-builder` and `potentially-emulated-builder` would be the most accurate, but are quite verbose... any other suggestions?

Reply:
maybe

```
build-platform-builder
target-platform-builder
```

?

Reply:
Reworded in [force-pushes](https://github.com/nextstrain/docker-base/compare/90c199a265138d5af3fb115f4a95dacc7b0d9f06..5478afcd2b3cfcbf72568b96be18f05dd8ecb643).

I opted for 

```
builder-build-platform
builder-target-platform
```

so the "builder" part of the image name wouldn't be so separated from "base" while keeping similar image and stage names (`nextstrain/base-build-platform-builder` vs. `nextstrain/base-builder-build-platform`). Still a bit verbose and "builder-build-platform" is a bit confusing, but I don't have anything better.

Reply:
@tsibley Once we settle on a good set of names here, I'll need you to create the respective repositories in the Nextstrain Docker Hub organization (and delete the old names) so the images can be pushed there.

Reply:
> Still a bit verbose and "builder-build-platform" is a bit confusing, but I don't have anything better.

Seems good enough to me, as far as warts go, since they're really only visible/matter to us.

I'll create the repos. (Didn't realize you couldn't!)

Reply:
https://hub.docker.com/u/nextstrain/base-builder-build-platform
https://hub.docker.com/u/nextstrain/base-builder-target-platform